### PR TITLE
Enhancement/Simplify Kaminari code

### PR DESCRIPTION
**Pull request by brunoocasali**
Remove unused methods in the `Pagination::Kaminari` class.

Also, a statement: 

I've spent a couple of hours porting the **Kaminari's `without_count`** (the countless mode) in the current pagination approach with our estimated pagination.
I could not make everything work because Kaminari did the `without_count` to work with AR collections, so to make it work would also imply that I would have ported Kaminari's code to a new adapter of plain arrays and then made it work with our adapter.

In other words, too much work for no real gain in real life.

If someone complains about the pagination performance, this will impact the engine's side more than ours.

### CODE REVIEW

File: lib/meilisearch/rails/pagination/kaminari.rb [hunk 1]
```diff
@@ -17,30 +17,18 @@ def initialize(array, options)
...
-          # noop
-          self
-        end
-
-        def offset(_num)
-          # noop
-          self
-        end
+        def self.create(results, total_hits, options = {})
+          offset = ((options[:page] - 1) * options[:per_page])
```
[the remark this answers is not in the document]
Comment (brunoocasali) [reply lib/meilisearch/rails/pagination/kaminari.rb:21]: The results will be empty, but using kaminari this only will be a problem if the user manually changed to page=0. (where the results will be empty).
